lib/src/utils.dart

```diff
@@ -124,16 +124,15 @@ String convertFromHString(int hstring) =>
 int convertToHString(String string) {
   final hString = calloc<HSTRING>();
   final stringPtr = string.toNativeUtf16();
+
   // Create a HSTRING representing the object
   try {
     final hr = WindowsCreateString(stringPtr, string.length, hString);
-    if (FAILED(hr)) {
-      throw WindowsException(hr);
-    } else {
-      return hString.value;
-    }
+    if (FAILED(hr)) throw WindowsException(hr);
+    return hString.value;
   } finally {
     free(stringPtr);
+    free(hString);
```

Comment:
Ohhh -- that's a brilliant catch. I missed that this is a Pointer<HSTRING> rather than the HSTRING itself.

Reply:
Actually, my prototype for #669 caught this :). I've made some progress on #669 but let's finish restructuring Win32 / WinRT first.